### PR TITLE
Documentation and Examples

### PR DESCRIPTION
**Pull request by codepitbull**
Documentation and Examples for vertx-lang-scala are now being generated.
Other modules will follow as I move along.

### CODE REVIEW

File: src/main/scala/io/vertx/lang/scala/ScalaApiTypeModel.scala [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+
+import java.util
+
+import io.vertx.codegen.`type`.{ClassKind, ParameterizedTypeInfo, TypeInfo}
+import io.vertx.codetrans.{CodeBuilder, CodeWriter, MethodSignature}
+import io.vertx.codetrans.expression.{ApiModel, ExpressionModel, MethodInvocationModel}
+
+import scala.collection.JavaConversions._
+
+class ScalaApiTypeModel(builder: CodeBuilder, expression: ExpressionModel) extends ApiModel(builder, expression) {
```
Comment (vietj) [inline src/main/scala/io/vertx/lang/scala/ScalaApiTypeModel.scala]: should be named `ScalaApiModel` , TypeModel` represents a class, not an instance

Comment (codepitbull) [inline src/main/scala/io/vertx/lang/scala/ScalaApiTypeModel.scala]: Done

File: pom.xml [hunk 2]
```diff
@@ -43,6 +43,13 @@
             <artifactId>vertx-core</artifactId>
         </dependency>
         <dependency>
+            <groupId>com.sun</groupId>
```
Comment (vietj) [inline pom.xml]: I think this is not necessary as codetrans depends on it already. So it depends on it via codetrans but the dependency is optional as side effect.

Comment (codepitbull) [inline pom.xml]: Fixed

File: src/main/scala/io/vertx/lang/scala/ScalaDocGenerator.scala [hunk 3]
```diff
@@ -0,0 +1,117 @@
+package io.vertx.lang.scala
```
Comment (jponge) [inline src/main/scala/io/vertx/lang/scala/ScalaDocGenerator.scala]: You should add a license header here.

File: src/main/scala/io/vertx/lang/scala/ScalaVerticle.scala [hunk 4]
```diff
@@ -3,50 +3,64 @@ package io.vertx.lang.scala
 import io.vertx.core.{AbstractVerticle, Future, Verticle}
```
Comment (jponge) [inline src/main/scala/io/vertx/lang/scala/ScalaVerticle.scala]: Same here and in most files, a license header would be great.

File: src/main/scala/io/vertx/lang/scala/ScalaDocGenerator.scala (hunk 3 above)
Comment (codepitbull) [inline src/main/scala/io/vertx/lang/scala/ScalaDocGenerator.scala]: fixed

File: src/main/scala/io/vertx/lang/scala/ScalaVerticle.scala (hunk 4 above)
Comment (codepitbull) [inline src/main/scala/io/vertx/lang/scala/ScalaVerticle.scala]: fixed

**Pull request merged**